src/llama.cpp

```diff
@@ -5067,6 +5097,9 @@ static void llm_load_vocab(
             } else if (
                 tokenizer_pre == "poro-chat") {
                 vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_PORO;
+            } else if (
+                tokenizer_pre == "jais") {
+                vocab.type_pre = LLAMA_VOCAB_PRE_TYPE_JAIS;
```

Comment:
Is it really necessary to add a new type of pre-tokenizer even if it is using the default implementation? I think it can be `default` / `LLAMA_VOCAB_PRE_TYPE_DEFAULT`. I am not sure what is the intended way to do this. Tagging @ggerganov.

Reply:
Either way works - not sure if it is better. We have already added different pretokenizer types for the same regexes in the case of StarCoder, Olmo, etc. so it's ok as it is.

However, based on https://huggingface.co/core42/jais-13b/raw/main/tokenizer.json I think this model should use the GPT-2 regexes, instead of the default:

https://github.com/ggerganov/llama.cpp/blob/c70d117c37cc7876e775d1e2722208a50c52edb3/src/llama.cpp#L13687-L13692